include/rocksdb/db.h

```diff
@@ -1051,6 +1051,11 @@ class DB {
     return IngestExternalFile(DefaultColumnFamily(), external_files, options);
   }
 
+  virtual Status IngestExternalFiles(
+      const std::vector<ColumnFamilyHandle*>& column_families,
+      const std::vector<std::vector<std::string>>& external_files,
+      const std::vector<IngestExternalFileOptions>& ingestion_options_list) = 0;
```

Comment:
1. Why do we need a IngestExternalFileOptions for every CF?
2. This kind of arguments always make me feel C-style, and I wonder whether it's the most intuitive way. One issue I personally feel uneasy is that we reply on users' code to make sure the same index of the vector represents to the same thing, and if they made a mistake, it's really hard to debug. Do we consider something like a struct with ColumnFamilyHandle* + file list, and a vector of this struct is passed in? In this way, we can avoid error handling of vector length discrepancy too.

Reply:
> 1. Why do we need a IngestExternalFileOptions for every CF?
Don't have a specific use case in mind, just trying to be general to support possible use cases.

> 2. This kind of arguments always make me feel C-style, and I wonder whether it's the most intuitive way. One issue I personally feel uneasy is that we reply on users' code to make sure the same index of the vector represents to the same thing, and if they made a mistake, it's really hard to debug. Do we consider something like a struct with ColumnFamilyHandle* + file list, and a vector of this struct is passed in? In this way, we can avoid error handling of vector length discrepancy too.
I guess we use something like
```
struct IngestFileArg {
  ColumnFamilyHandle* cfh,
  const std::vector<std::string>& external_files;
  const IngestExternalFileOptions& ingestion_options;
  ...
};
```

Reply:
Hmm, I looked at what's in IngestExternalFileOptions. It is true that, they can all implemented per CF, kind of. On the other hand, once as a part of API, this kind of flexibility needs to be supported forever. You can make a call.

Reply:
+1 to vector<IngestFileArg>. I don't think we need that kind of flexibility for different ingestion_options per CF though (but it doesn't hurt either :))

Reply:
@siying having a vector of the structs you mentioned still relies on the user populating the vector correctly with each column family associated with the right list of external SST files. If they make a mistake, I think it's equally difficult to diagnose. Maybe I am missing something?

Reply:
@riversand963 There are less chances of things going wrong (no size mismatch checking, no missed push_back messing up the rest of the arguments, etc), and the intention is more explicit. Yes, user can still make mistakes - but a good API design should make it really easy to do the right thing, and harder to do the wrong thing. Either way, I think a sanity check for the SST belong to the right column family is still good to have.

Reply:
@yizhang82 I think the check is here: https://github.com/facebook/rocksdb/blob/master/db/external_sst_file_ingestion_job.cc#L48

Reply:
@riversand963 Thanks. I was simply commenting from API design perspective 😄 It's good to confirm the check is there.